### PR TITLE
Feature/overlay refactor

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Use Heroes in the complex bottom sheet
- Add empty state placeholder

### CODE REVIEW

File: src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.scss
Comment (joebochill) [inline src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.scss:6]: Can we pull this from the colors package

File: src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html
Comment (joebochill) [inline src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html:9]: The whole hero should be clickable, rather than just the icon
- You'll probably want to convert those mat-icon-buttons to plain mat-icons when you make this change

File: src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html
Comment (joebochill) [inline src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html:3]: I think you got the labels backwards...Huayun wants us to _use_ the Overline label style in both React and Angular.

**Review by joebochill — CHANGES_REQUESTED** [review]

File: src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html
Comment (emclaug2) [reply src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html:3]: Ah, okay! This is reverted now.

File: src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html
Comment (emclaug2) [reply src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.html:9]: I've made the entire Hero clickable; no more mat-buttons.

File: src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.scss
Comment (emclaug2) [reply src/app/pages/overlays/complex-bottom-sheet/bottom-sheet/bottom-sheet.scss]: Absolutely :)

**Comment by emclaug2** [comment]
Once this PR gets merged I'll update React to match.

**Comment by jeffvg** [comment]
The title here is in h3 tag but is rendering in all uppercase and only noticed because the cypress test is failing on it.

image

**Comment by emclaug2** [comment]
@jeffvg  I've removed the  `text-transform: uppercase` from the overline typography.

This is a comparison between the two overlay examples right now:

image

I'm planning on updating React to do the following:
1. match angular tyopgraphy
2.  Use hero components
3.  Match container padding
4.  Update bottom button to have elevation

**Comment by joebochill** [comment]
Both should be using the `overline` style with uppercase transform for the labels.

@huayunh I warned you that change was going to cause some confusion lol 😆

**Comment by emclaug2** [comment]
The heroes now look a bit spaced out, so the fix would be to apply a max-width to each hero.  I'm going to hold off on applying that style for now since the root class of the hero is going to be changed as a part of another PR.

**Comment by jeffvg** [comment]
> @jeffvg I've removed the `text-transform: uppercase` from the overline typography.
> 
> This is a comparison between the two overlay examples right now:
> 
> React is on the left, angular is on the right.
> image
> 
> I'm planning on updating React to do the following:
> 
> 1. match angular typography
> 2. Use hero components
> 3. Match overlay container padding
> 4. Update bottom button to have elevation

did that removal of text-transform: uppercase get pushed in?

**Comment by emclaug2** [comment]
> > @jeffvg I've removed the `text-transform: uppercase` from the overline typography.
> > This is a comparison between the two overlay examples right now:
> > React is on the left, angular is on the right.
> > image
> > I'm planning on updating React to do the following:
> > 
> > 1. match angular typography
> > 2. Use hero components
> > 3. Match overlay container padding
> > 4. Update bottom button to have elevation
> 
> did that removal of text-transform: uppercase get pushed in?



> > @jeffvg I've removed the `text-transform: uppercase` from the overline typography.
> > This is a comparison between the two overlay examples right now:
> > React is on the left, angular is on the right.
> > image
> > I'm planning on updating React to do the following:
> > 
> > 1. match angular typography
> > 2. Use hero components
> > 3. Match overlay container padding
> > 4. Update bottom button to have elevation
> 
> did that removal of text-transform: uppercase get pushed in?

Yep, the "Sort By" and "Show" text is now upper-cased.

**Comment by joebochill** [comment]
> The heroes now look a bit spaced out, so the fix would be to apply a max-width to each hero. I'm going to hold off on applying that style for now since the root class of the hero is going to be changed as a part of another PR.

That max-width would go in the example though, not in the component, right? We don't know what kind of labels people will be using, so we should probably leave that up to them.

**Comment by joebochill** [comment]
Refactor looks good. Just one question, should we be using black[500] for the hero icons (in this particular use case)?

**Comment by emclaug2** [comment]
> > The heroes now look a bit spaced out, so the fix would be to apply a max-width to each hero. I'm going to hold off on applying that style for now since the root class of the hero is going to be changed as a part of another PR.
> 
> That max-width would go in the example though, not in the component, right? We don't know what kind of labels people will be using, so we should probably leave that up to them.

You're right, I didn't make the change yet because the class names were going to change soon, but I can just apply css to the hero tags instead of the classes.

**Comment by emclaug2** [comment]
Here's the complex bottomsheet now:

image

**Comment by emclaug2** [comment]
> Refactor looks good. Just one question, should we be using black[500] for the hero icons (in this particular use case)?

I'm not sure; right now it's using the default Hero colors which I think looks alright.

**Comment by joebochill** [comment]
> > Refactor looks good. Just one question, should we be using black[500] for the hero icons (in this particular use case)?
> 
> I'm not sure; right now it's using the default Hero colors which I think looks alright.

I'm wondering because the other text in the bottomsheets is black[500]

@huayunh thoughts?

**Comment by emclaug2** [comment]
@joebochill  @huayunh 
Is the elevation above the cancel button area preferred over a flat divider?   This is only present on mobile viewports. 

image

**Comment by huayunh** [comment]
(1) Happy with the default hero

(2) the elevation is there so that the cancel button is pinned at the bottom, and we can scroll vertically when the content overflows. This overflow typically only occurs on a mobile device held at the landscape position.

**Review by jeffvg — APPROVED** [review]

**Review by joebochill — APPROVED** [review]

**Pull request merged**
